job.js

```diff
@@ -357,6 +377,57 @@ module.exports = function job (options) {
       }
     })
   }
+
+  let updateRethinkJob = async function (qdata) {
+    return new Promise(async(resolve, reject) => {
+      try {
+        // check port number range
+        checkPortNumber(newoptions.connection.port)
+        .then(async result => {
+          let queueObj = await createJobQueue(newoptions.connection, newoptions.queue)
+          queueObj.on('error', (err) => {
+            // err object is system error
+            reject(customError(err))
+          })
+
+          await updateJob(queueObj, qdata.findVal, qdata.data, qdata.options)
+            .then(result => {
+              resolve(result)
+            })
+            .catch(err => { customError(err) })
+        })
+        .catch(err => {
+          // this is return custom error
+          reject(err)
+        })
+        // let dbDriver = await createRethinkdbDash(options.connection)
+      } catch (err) {
+        // some system fatal error happend - no way to recover !!
+        reject(customError(err))
+      }
+    })
+  }
+
+  let updateJob = async function (queueObj, val, jobData, options) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        queueObj.getJob(val, true).then(async (jobs) => {
+          if(options) {
+            for (key in options) {
+              jobs[0][key] = options[key]
+            }
+          }
+          jobs[0].data = jobData
+          await jobs[0].update()
+          jobs = jobs.map(function (a) { delete a.q; return a })
+          resolve(jobs)
+        }).catch(err => reject(err))
+      } catch (err) {
+        reject(err)
+      }
+    })
+  }
+
 }
 
 let customError = function (err, errorCode) {
```

Comment:
set invalid parameter error key
date enable error

Reply:
requested changes are addressed in new commit ***'requested changes in review'*** .
**err.name** is now used to identify the error.